### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by TiamHeydari** [CellProfiler/CellProfiler#4514]
Hi, 
I am using Linux 20.xx
I get this error:

The conflict is caused by:
    cellprofiler 4.2.1 depends on pyzmq~=22.3
    cellprofiler-core 4.2.1 depends on pyzmq==18.0.1

Thanks,

**Comment by bethac07** [comment]
You can also avert this by also running cellprofiler-core from source rather than from a wheel.  Apologies for the inconvenience!

**Comment by nicost** [comment]
Following the Linux installation instructions (```
git clone https://github.com/CellProfiler/CellProfiler.git 
cd CellProfiler 
pip3 install .
```), I run into the same issue:

```
ERROR: Cannot install CellProfiler and cellprofiler==4.2.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    cellprofiler 4.2.1 depends on h5py~=3.6.0
    cellprofiler-core 4.2.1 depends on h5py==3.2.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```

How do I "run cellprofiler-core" from source rather than a wheel?

**Comment by Jiraya-Aizen** [comment]
Hi, I have been trying to install cellprofiler in anaconda3 virtual environment, but it shows that the cellprofiler version is not found. I have tried with different versions, still the error remains. Can someone help?
![Capture](https://user-images.githubusercontent.com/52849593/159456753-8d37af84-eedb-4c81-a5f8-d47875fa50d5.PNG)

**Comment by braymp** [comment]
@nicost : In addition to my hacky approach above, I think you can also solve the issue by checking out the 4.2.1 tag after cloning:
`git checkout tags/4.2.1`
The original tagged version should match the cellprofiler-core version.

**Comment by Jiraya-Aizen** [comment]
@braymp I have tried with versions starting from 4.2.1 all the way to 4.0.1. The issue pertains.

**Comment by omsai** [comment]
The build fails with python 3.9, so strictly speaking cellprofiler 4.2.1 which is listing as supporting up to python 3.8 is valid.

For my python 3.9 on Debian 11 Bullseye, I can install cellprofiler after removing the pinned versions for zmq and wx:

```bash
# Install patched cellprofiler-core 4.2.1
git clone https://github.com/CellProfiler/core CellProfiler-core
cd CellProfiler-core/
git checkout tags/v4.2.1
# Remove hardcoded versions to use system / current wheels.
sed -E -i \
    -e 's#pyzmq==[^"]+#pyzmq#' \
    -e 's#wxPython==[^"]+#wxPython#' \
    setup.py
git diff
python3 -m pip install --user .

# Install patched cellprofiler 4.2.1
cd ../
git clone https://github.com/CellProfiler/CellProfiler.git
cd CellProfiler/
git checkout tags/v4.2.1
# Remove hardcoded versions to use system / current wheels.
sed -E -i \
    -e 's#pyzmq==[^"]+#pyzmq#' \
    -e 's#wxPython==[^"]+#wxPython#' \
    setup.py
git diff
python3 -m pip install --user .
```

Yes, I confirmed this doesn't break anything because most of the tests pass except for database and volume measurement things and a few edge which don't seem relevant to me:

```bash
# Run cellprofiler tests:
python3 -m pip install --user --upgrade pytest
# Workaround for "ValueError: module functions cannot set METH_CLASS or METH_STATIC"
python3 -m pip install --user --upgrade matplotlib
pytest
# 53 failed, 1387 passed, 29 skipped, 9133 warnings in 288.87s (0:04:48)
# ^ Holy flood of warnings, Batman!
pytest -q -p no:warnings -k "not database and not volume" --ignore='tests/modules/test_exporttodatabase.py'
# 11 failed, 1162 passed, 16 skipped, 205 deselected in 216.63s (0:03:36)
```

The system dependencies I installed along the way are:

```bash
  sudo apt install \
       libmariadb-dev \
       openjdk-11-jre-headless \
       libgtk-3-dev \
       libzmq3-dev \
       python3-wxgtk4.0
```

wxPython folks recommend installing from apt instead of pypi, and I'm not sure if libzm13-dev was necessary for the wheel.

By the way, in general I feel that too many dependency versions are pinned with no clear reason such as a reference to a GitHub issue or StackOverflow discussion.  I understand that pins help setup a developer environment, but is a pain for us users and actually adds more maintenance burden to you.  For example, why is pytest pinned to <4?
